### DISCUSSION TITLE
Unable to run VS Code outside the Developer Command Prompt

### DISCUSSION THREAD

**Discussion by clicktodev** [microsoft/vscode-cpptools#12334] [Q&A]
According to the official docs, I should be able to start vscode inside developer command prompt mode with a simple config (no need to launch vscode from the developer command prompt manually).
https://code.visualstudio.com/docs/cpp/config-msvc#_run-vs-code-outside-the-developer-command-prompt
For some reason, this isn't working at all.
one thing I tried, is create a custom terminal profile.

vscode global settings file
```json
	"terminal.integrated.profiles.windows": {
		"vs-compiler": {
			"path": "C:\\Windows\\System32\\cmd.exe",
			"args": [
				"/k",
				"C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/Tools/VsDevCmd.bat"
			]
		}
	},
```
local workspace settings file:
```json
{
	"terminal.integrated.defaultProfile.windows": "vs-compiler",
}

```

this works only if i open the terminal and type out the compile command, but it's not working when I run the `run c/c++ file` command provided by the c/c++ extension.

workspace `tasks.json`
```json
{
	"version": "2.0.0",
	"windows": {
		"options": {
			"shell": {
				"executable": "C:\\Windows\\System32\\cmd.exe",
				"args": [
					"/k",
					"C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/Tools/VsDevCmd.bat"
				]
			}
		}
	},
	"tasks": [
		{
			"type": "shell",
			"label": "cl.exe build active file",
			"command": "cl.exe",
			"args": [
				"/Zi",
				"/EHsc",
				"/Fe:",
				"${fileDirname}\\${fileBasenameNoExtension}.exe",
				"${file}"
			],
			"problemMatcher": ["$msCompile"],
			"group": "build"
		},
		{
			"type": "cppbuild",
			"label": "C/C++: cl.exe build active file",
			"command": "cl.exe",
			"args": [
				"/Zi",
				"/EHsc",
				"/nologo",
				"/Fe${fileDirname}\\${fileBasenameNoExtension}.exe",
				"${file}"
			],
			"options": {
				"cwd": "${fileDirname}"
			},
			"problemMatcher": ["$msCompile"],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "Task generated by Debugger."
		}
	]
}
```

**Comment by jon-zu** [comment]
I'm facing a similar issue as I want to remote on a windows host remotely via SSH.  It's incredible frustrating that this doesn't seem to work, as It seems to be a very common use-case for remote development on a windows host machine from linux/osx. I've already wasted over too much time researching this, just to be able to debug some tests on Windows. No other toolchain/language I've used is this miserable to set up for remote development.

**Comment by sean-mcmanus** [comment]
The "args" should be similar to

```
        "args": [
          "/C",
          // The path to VsDevCmd.bat depends on the version of Visual Studio you have installed.
          "\"C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/Tools/VsDevCmd.bat\"",
          "&&"
        ]
```
